functional/sev/run.sh

```diff
@@ -0,0 +1,337 @@
+
+export KBS_DB_HOST="127.0.0.1"
+export KBS_DB_USER="root"
+export KBS_DB_PW="root"
+export KBS_DB="simple_kbs"
+export KBS_DB_TYPE="mysql"
+
+esudo() {
+  sudo -E PATH=$PATH "$@"
+}
+
+modify_kata_config() {
+  esudo mkdir -p /etc/kata-containers/
+  esudo install -o root -g root -m 0640 /usr/share/defaults/kata-containers/configuration.toml /etc/kata-containers
+
+  # Enable full debug
+  esudo sed -i -e 's/^# *\(enable_debug\).*=.*$/\1 = true/g' /etc/kata-containers/configuration.toml
+  
+  # kernel, initrd, image, firmware
+  esudo sed -i -e 's/kernel = "\(.*\)"/kernel = "\/usr\/share\/kata-containers\/vmlinuz-sev.container"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/initrd = "\(.*\)"/initrd = "\/usr\/share\/kata-containers\/kata-containers-initrd.img"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/image = "\(.*\)"/#image = ""\ninitrd = "\/usr\/share\/kata-containers\/kata-containers-initrd.img"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e "s|^\(firmware = \"\).*\"|\1${EDK2_REPO_DIR}/Build/AmdSev/DEBUG_GCC5/FV/OVMF.fd\"|g" /etc/kata-containers/configuration.toml
+  
+  # kernel_params
+  esudo sed -i -e 's/kernel_params = "\(.*\)"/kernel_params = " agent.log=debug initcall_debug agent.config_file=\/etc\/default\/agent.cfg console=ttyAMA0,115200 console=tty highres=off console=ttyS0"/g' /etc/kata-containers/configuration.toml
+
+  # confidential_guest, guest_attestation
+  esudo sed -i -e 's/# confidential_guest = true/confidential_guest = true/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/#guest_pre_attestation = true/guest_pre_attestation = true/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/#guest_pre_attestation_proxy="localhost:50051"/guest_pre_attestation_proxy = "localhost:44444"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/#guest_pre_attestation_keyset="KEYSET-1"/guest_pre_attestation_keyset = "KEYSET-1"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e "s|^#\(guest_pre_attestation_secret_guid=\".*\"$\)|\1|g" /etc/kata-containers/configuration.toml
+  esudo sed -i -e "s|^#\(guest_pre_attestation_secret_type=\".*\"$\)|\1|g" /etc/kata-containers/configuration.toml
+  esudo sed -i -e "s|^#\(sev_cert_chain=\".*\"$\)|\1|g" /etc/kata-containers/configuration.toml
+
+  # shared_fs, seccomp
+  esudo sed -i -e 's/shared_fs = "\(.*\)"/shared_fs = "virtio-9p"/g' /etc/kata-containers/configuration.toml
+  esudo sed -i -e 's/#disable_guest_seccomp=true/disable_guest_seccomp=true/g' /etc/kata-containers/configuration.toml
+}
+
+initrd_add_files() {
+  rootfs_dir="$(mktemp -d)"
+  initrd_path="$(kata-runtime kata-env --json | jq -r .Initrd.Path)"
+  
+  # Extract gzip initrd into temp directory
+  zcat "${initrd_path}" | cpio --extract --preserve-modification-time --make-directories --directory="${rootfs_dir}"
+  
+  # Copy agent.cfg to initrd
+  esudo cp "functional/sev/fixtures/agent.cfg" "${rootfs_dir}/etc/default"
```

Comment:
Maybe we should parameterize the agent config path since we also have to add it to the kerne params? Also, is this path in `/etc/default` consistent with the TDX tests?

Reply:
@fidencio are you guys putting the agent.cfg in a different place. I was looking through the codebase and was having trouble finding where you guys are doing this.